test/jdk/java/time/test/java/time/zone/TestZoneRulesProvider.java

```diff
@@ -91,4 +78,62 @@ protected NavigableMap<String, ZoneRules> provideVersions(String zoneId) {
             }
         });
     }
+
+    /**
+     * Tests whether registering a provider twice will still leave it registered.
+     * @bug 8302983
+     */
+    @Test
+    public void test_registerTwice() {
+        String zone = "MyID";
+        var provider = new IdsOnlyZoneRulesProvider(Set.of(zone));
+        assertFalse(ZoneId.getAvailableZoneIds().contains(zone), "Unexpected availability for " + zone);
+        ZoneRulesProvider.registerProvider(provider);
+        assertTrue(ZoneId.getAvailableZoneIds().contains(zone), "Unexpected non-availability for " + zone);
+        try {
+            ZoneId.of(zone);
+        } catch (ZoneRulesException e) {
+            fail("ZoneId instance for " + zone + " should be obtainable");
+        }
+
+        try {
+            ZoneRulesProvider.registerProvider(provider);
+            fail("Registering an already registered provider should throw an exception");
+        } catch (ZoneRulesException e) {
+            // Ignore. Failure on duplicate registration is expected.
+        }
+
+        // availability check
+        assertTrue(ZoneId.getAvailableZoneIds().contains(zone), "Unexpected non-availability for " + id);
```

Comment:
Did you actually run the test??? It still does not compile. Please make sure you run the test under the modified jdk and passes.

```
STDERR:
/Users/nsato/dev/jdk/git/master/open/test/jdk/java/time/test/java/time/zone/TestZoneRulesProvider.java:107: error: cannot find symbol
        assertTrue(ZoneId.getAvailableZoneIds().contains(zone), "Unexpected non-availability for " + id);
                                                                                                     ^
  symbol:   variable id
  location: class TestZoneRulesProvider
```